src/components/_input.scss

```diff
@@ -0,0 +1,103 @@
+@use "../design-tokens/color";
+@use "../design-tokens/typography";
+
+// ----------
+//  Block
+// ----------
+
+@mixin Manifold__Input {
+  transition: border-color 150ms linear;
+
+  // ----------
+  //  Elements
+  // ----------
+
+  & label {
+    @include typography.Manifold__Typography__SubheadingSmall;
+
+    display: block;
+    margin-bottom: 0.25rem;
+    color: color.$color-grayDark;
+    transition-timing-function: linear;
+    transition-duration: 150ms;
+    transition-property: color;
+  }
+
+  & input {
+    @include typography.Manifold__Typography__SmallScreenBody;
+
+    display: block;
+    box-sizing: border-box;
+    width: auto;
+    min-width: none;
+    padding: 0.5em 0.75em;
+    color: color.$color-grayDarker;
+    background: color.$color-white;
+    border: 1px solid color.$color-utilitiesGray;
+    border-radius: 3px;
+    outline: 0;
+    transition-timing-function: linear;
+    transition-duration: 150ms;
+    transition-property: border-color;
+
+    @media (min-width: 600px) {
+      @include typography.Manifold__Typography__Body;
+    }
+
+    &[type='number'],
+    &[inputmode='numeric'],
+    &[pattern='[0-9]*'] {
+      @include typography.Manifold__Typography__MonoBody;
+    }
+  }
+
+  & input
+
+  // ----------
+  //  States
+  // ----------
+
+  & input:invalid,
+  & input[aria-invalid] {
+    border-color: color.$color-red;
+    box-shadow: none;
+  }
+
+  & input:disabled {
+    color: color.$color-gray;
+    background: color.$color-grayLightest;
+  }
```

Comment:
With `disabled` and `aria-invalid`, we were _juuust_ able to eke out a way to control the styling from a top-level JS framework. Without these standard attributes, I don’t know what we’d do (probably abuse `data-*` attributes)